### PR TITLE
[cmd/telemetrygen] Better defaults for HTTP exporter mode

### PR DESCRIPTION
**Pull request by jmsnll**
**Description:**

- Refactored and moved `otlp-http-url-path` flag from the common flags for each command
    - Added flag to the `traces` command with the default `/v1/traces`
    - Added flag to the `metrics` command with the default `/v1/metrics`
    - Flag was not used for the `logs` command so no longer exposed
- Handle changing the default endpoint based on the communication mode (gRPC or HTTP)
    - Flag `--otlp-endpoint` now defaults to empty string and targets a new attribute `CustomEndpoint` on `common.Config`
    - Added the `Endpoint()` getter function to the `common.Config` struct for retrieving the correct endpoint
    - When `CustomEndpoint` is empty then either `DefaultGRPCEndpoint` or `DefaultHTTPEndpoint` are chosen based upon the value of `config.UseHTTP` 
- **Misc**:  Split out the creation of trace/metric exporters into standalone factory functions with docstrings available in `exporters.go` in both modules 
- **Misc**: Removed the "default value is" from the descriptions of some flags as it was repeated information

**Link to tracking Issue:** #26999

**Testing:** 

Added new set of unit tests to cover the addition of the `config.Endpoint()` getter. 

Running `telemetrygen traces --otlp-http --otlp-insecure` now correctly sends traces using HTTP to the default HTTP endpoint. 

**Documentation:** 

No documentation added/changed but updated some command flag descriptions.

### CODE REVIEW

File: cmd/telemetrygen/internal/metrics/config.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ func (c *Config) Flags(fs *pflag.FlagSet) {
 	c.MetricType = metricTypeGauge
 
 	c.CommonFlags(fs)
+
+	fs.StringVar(&c.HTTPPath, "otlp-http-url-path", "/v1/metrics", "Which URL path to write to")
```
Comment (codeboten) [inline cmd/telemetrygen/internal/metrics/config.go:26]: please add a test for this parameter

File: cmd/telemetrygen/internal/traces/config.go [hunk 2]
```diff
@@ -23,6 +23,9 @@ type Config struct {
 // Flags registers config flags.
 func (c *Config) Flags(fs *pflag.FlagSet) {
 	c.CommonFlags(fs)
+
+	fs.StringVar(&c.HTTPPath, "otlp-http-url-path", "/v1/traces", "Which URL path to write to")
```
Comment (codeboten) [inline cmd/telemetrygen/internal/traces/config.go:27]: likewise, would be great to have a test for this as well

**Review by codeboten — COMMENTED** [review]
Thanks for the PR @jmsnll, could the path be added for logs as well?

File: cmd/telemetrygen/internal/metrics/config.go (hunk 1 above)
Comment (jmsnll) [reply cmd/telemetrygen/internal/metrics/config.go:26]: certainly, would a test checking if the exporter is created with `otlptracehttp.WithURLPath(cfg.HTTPPath)` be suffice?

**Comment by jmsnll** [comment]
And thanks for the review @codeboten! 

I could be wrong here, but Isn't the URL path only relevant for HTTP? I can add it to the the `logs` command but since it's only using gRPC at the moment I don't see where the value would be actually used.

File: cmd/telemetrygen/internal/metrics/config.go (hunk 1 above)
Comment (codeboten) [reply cmd/telemetrygen/internal/metrics/config.go:26]: Yup! that works

**Comment by codeboten** [comment]
> I could be wrong here, but Isn't the URL path only relevant for HTTP? I can add it to the the logs command but since it's only using gRPC at the moment I don't see where the value would be actually used.

Of course... i forgot that logs aren't supported for http yet 🤦🏻 ignore my comment 👍🏻

File: cmd/telemetrygen/internal/metrics/config.go (hunk 1 above)
Comment (jmsnll) [reply cmd/telemetrygen/internal/metrics/config.go:26]: I gave it a shot but ran into an issue with most of the attributes and structs I needed for writing the test being internal to the `otlp{trace,metrics}{grpc,http}` packages (like the option functions and the config structs they're applied to).

I'm maybe missing something with the above, but the best I was able to come up with was a test I've added in `config_test.go` which just simply  checks the defaults are different per command.

**Review by mx-psi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by evan-bradley** [comment]
@jmsnll Could you fix the linting errors?

**Comment by codeboten** [comment]
@evan-bradley this is my bad... tried to resolve a conflict via the github UI... will fix

**Pull request merged**